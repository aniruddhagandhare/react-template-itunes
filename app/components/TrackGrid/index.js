/**
 *
 * TrackGrid
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, Tag, Tooltip } from 'antd';
import { Link } from 'react-router-dom';
import For from '@components/For';
import T from '@components/T';
import { propTypeConstants } from '@utils/propTypeConstants';
import { colors, fonts } from '@app/themes/';

const GridCard = styled.div`
  display: grid;
  margin-top: 2em;
  grid-template-columns: repeat(auto-fit, minmax(18.75em, 1fr));
  gap: 1.5em;
`;
const FlexContainer = styled.div`
  display: flex;
  align-items: center;
  img {
    border-radius: 5em;
    width: 4em;
    height: 4em;
  }
`;
const StyledT = styled(T)`
  && {
    ${props => props.small && fonts.size.small()};
  }
`;
const VSpace = styled.div`
  margin: 0.5em 0;
`;
const Container = styled.div`
  width: 100%;
  padding: 1em 0;
`;
const Content = styled.div`
  margin-left: 2em;
  flex: 1;
`;

function TrackGrid({ songs }) {
  return (
    <Container data-testid="track-grid">
      <For
        of={songs}
        isRow={false}
        ParentComponent={GridCard}
        renderItem={(song, idx) => (
          <Card key={idx}>
            <FlexContainer>
              <img src={song.artworkUrl60} />
              <Content>
                <Tooltip
                  placement="topLeft"
                  title={<StyledT small id="collection_name" values={{ collectionName: song.collectionName }} />}
                >
                  <Link to={`/track/${song.trackId}`} style={{ color: colors.primary }}>
                    <StyledT id="track_name" values={{ trackName: song.trackName }} />
                  </Link>
                </Tooltip>
                <StyledT
                  small
                  id="artist_name"
                  type="subtext"
                  style={{ color: colors.text }}
                  values={{ artistName: song.artistName }}
                />
                <VSpace />
                <Tag color={colors.primary}>
                  <StyledT small id="genre_name" values={{ genreName: song.primaryGenreName }} />
                </Tag>
              </Content>
            </FlexContainer>
          </Card>
        )}
      />
    </Container>
  );
}

TrackGrid.propTypes = {
  songs: PropTypes.arrayOf(propTypeConstants).isRequired
};

export default memo(TrackGrid);
