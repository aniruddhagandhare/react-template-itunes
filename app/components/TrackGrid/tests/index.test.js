/**
 *
 * Tests for TrackGrid
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import TrackGrid from '../index';

describe('<TrackGrid />', () => {
  let songs;
  beforeEach(() => {
    songs = [
      {
        artworkUrl: '',
        previewUrl: '',
        collectionName: '',
        trackName: '',
        artistName: '',
        trackPrice: 0,
        trackExplicitness: '',
        trackPrimaryGenreName: '',
        trackViewUrl: ''
      }
    ];
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<TrackGrid songs={songs} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 TrackGrid component', () => {
    const { getAllByTestId } = renderWithIntl(<TrackGrid songs={songs} />);
    expect(getAllByTestId('track-grid').length).toBe(1);
  });
});
