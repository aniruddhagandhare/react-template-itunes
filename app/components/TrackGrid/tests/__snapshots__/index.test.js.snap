// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackGrid /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="TrackGrid__Container-sc-1fqyrr9-4 glDrzJ"
      data-testid="track-grid"
    />
  </div>
</body>
`;
